bugbug/test_scheduling.py

```diff
@@ -310,21 +310,62 @@ def get_test_scheduling_history(granularity):
         yield obj["revs"], obj["data"]
 
 
-def get_past_failures(granularity, readonly):
-    if granularity == "label":
-        past_failures_db = os.path.join("data", PAST_FAILURES_LABEL_DB)
-    elif granularity == "group":
-        past_failures_db = os.path.join("data", PAST_FAILURES_GROUP_DB)
-    elif granularity == "config_group":
-        past_failures_db = os.path.join("data", PAST_FAILURES_CONFIG_GROUP_DB)
-    else:
-        raise UnexpectedGranularityError(granularity)
+class PastFailures:
+    def __init__(self, granularity, readonly):
+        if granularity == "label":
+            past_failures_db = os.path.join("data", PAST_FAILURES_LABEL_DB)
+        elif granularity == "group":
+            past_failures_db = os.path.join("data", PAST_FAILURES_GROUP_DB)
+        elif granularity == "config_group":
+            assert False, "config_group granularity not supported for past failures"
+        else:
+            raise UnexpectedGranularityError(granularity)
+        self.granularity = granularity
 
-    return shelve.Shelf(
-        LMDBDict(past_failures_db[: -len(".tar.zst")], readonly=readonly),
-        protocol=pickle.DEFAULT_PROTOCOL,
-        writeback=not readonly,
-    )
+        self.db = shelve.Shelf(
+            LMDBDict(past_failures_db[: -len(".tar.zst")], readonly=readonly),
+            protocol=pickle.DEFAULT_PROTOCOL,
+            writeback=not readonly,
+        )
+
+    @property
+    def push_num(self) -> int:
+        return self.db["push_num"]
+
+    @push_num.setter
+    def push_num(self, value: int) -> None:
+        self.db["push_num"] = value
```

Comment:
Why do we need setters? Can the values be passed to the constructor instead?

(same for `all_runnables`)

Reply:
The values are being changed, for example here: https://github.com/mozilla/bugbug/pull/3691/files#diff-67e71ee19f5db00669e94c709cb6de177b6117efc0bfbcf96b46e365e7433ef5R340.

Reply:
In this specific case, the value didn't change based on any external factor. Which can be determined in the constructor.

Reply:
Never mind, I see it is modified in https://github.com/mozilla/bugbug/pull/3691/files#diff-f3557fe3a5b0819ec4d3a2f85a607a9450e4b9f18168838e95d3d2672564f8b4R566